### PR TITLE
login page

### PR DESCRIPTION
**Pull request by Fir0w**

### CODE REVIEW

**Comment by Fir0w** [comment]
ezgif com-video-to-gif

File: server/index.js [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
+
+connectDB();
 
 const app = express();
 
-app.get('/', (req, res) => {
-    res.send('<h1>Hello world!</h1>');
+//Middleware
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: true }));
```
Comment (VicDosya) [inline server/index.js:13]: I think instead of body-parser, I think we can do all this with express (without bodyParser)
for example:
app.use(express.json());
app.use(express.urlencoded({ extended: false }));

Can you confirm this pls? If this works, we can remove body-parser (and uninstall it) :)

File: client/src/App.css [hunk 2]
```diff
@@ -5,7 +5,7 @@
   text-align: center;
 }
 
-.logo {
+/* .logo {
```
Comment (VicDosya) [inline client/src/App.css:8]: Remove comments from this file

File: client/components/Login.jsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                    <form onSubmit={onSubmitHandler}>
+                        <label htmlFor="email">
+                            <span>Email: </span>
+                            <input value={inputEmail} type="email" id="email" placeholder='Email' autoComplete='on' onChange={(e) => setInputEmail(e.target.value)} onBlur={(e) => validateEmail(e.target.value)} />
+                            {validEmail && <p className={styles.invalid}>Invalid email</p>}
+                        </label>
+                        <label htmlFor="password">
+                            <span>Password: </span>
+                            <input value={inputPassword} type="password" id="password" placeholder='Password' autoComplete='on' minLength={8} maxLength={16} onChange={(e) => setInputPassword(e.target.value)} onBlur={(e) => validatePassword(e.target.value)} />
+                            {validPassword && <p className={styles.invalid}>Invalid Password</p>}
```
Comment (VicDosya) [inline client/components/Login.jsx:61]: I don't think we should let client know that the password is invalid when he tries to login. 
it's not commonly shown in my opinion? what do you think ?

File: client/components/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                <h2>Post React App</h2>
+            </a>
+            <div className={styles.loginContainer}>
+                <h1>Login</h1>
+                <div className={styles.inputContainer}>
+                    <form onSubmit={onSubmitHandler}>
+                        <label htmlFor="email">
+                            <span>Email: </span>
+                            <input value={inputEmail} type="email" id="email" placeholder='Email' autoComplete='on' onChange={(e) => setInputEmail(e.target.value)} onBlur={(e) => validateEmail(e.target.value)} />
+                            {validEmail && <p className={styles.invalid}>Invalid email</p>}
```
Comment (VicDosya) [inline client/components/Login.jsx:56]: the name `validEmail` is confusing ? 
so if `validEmail` then it should show Invalid email ?

should it be `invalidEmail` ?

Same goes with the password.

File: client/components/Login.jsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+    const postResponse = async () => {
+        const formData = { email: inputEmail, password: inputPassword }
+        try {
+            const res = await axios.post('/api/login', formData);
+            console.log(res);
+        } catch (err) { console.log(err); }
+    };
+
+    const validateEmail = (email) => {
```
Comment (VicDosya) [inline client/components/Login.jsx:26]: I see a repeating pattern between `validateEmail` and `validatePassword` .
Do you think we can turn them into one function and reuse it for email and password ? WDYT ?

File: client/components/Login.jsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    const [validEmail, setValidEmail] = useState(false);
+    const [validPassword, setValidPassword] = useState(false);
+
+
+    const onSubmitHandler = (event) => {
+        event.preventDefault();
+        postResponse();
+    };
+
+    const postResponse = async () => {
```
Comment (VicDosya) [inline client/components/Login.jsx:18]: when user presses the submit button, it fires this function.
But there's a small problem, we are not validating the inputs here!

If i understand correctly, you are validating using `onBlur` ( which is great! but is it necessary ?)

I think a better approach would be to validate input fields INSIDE the `postResponse` function so that we won't accidentally send invalid input values to the server.

So basically, if ALL input fields are VALID -> send server response. if NOT valid then reveal an error message :)

File: client/components/Index.jsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        
+    // };
+
+    const navigate = useNavigate();
+
+    
+    return (
+        <>
+            <h1>Welcome to React Post App</h1>
+            <img className={styles.logo} src="/logo.png" alt="" />
```
Comment (VicDosya) [inline client/components/Index.jsx:27]: pls don't keep `alt` to be empty.

**Review by VicDosya — CHANGES_REQUESTED** [review]
Great work!! left some comments for you :)

File: client/components/Login.jsx (hunk 4 above)
Comment (VicDosya) [reply client/components/Login.jsx:56]: or if you want to keep it then let it be something like:
!validEmail && ..."invalid email"

File: server/index.js (hunk 1 above)
Comment (Fir0w) [reply server/index.js:13]: it works
